### ISSUE TITLE
[Android] Recording of WAV, OPUS and other files fails silently on certain device(s)

### ISSUE THREAD

**Issue by felixmede** [llfbandit/record#393]
**Package version**
- 1.2.5
- 1.2.6

**Environment**
 - OS: Android 11
 - ONEPLUS A6013

**Describe the bug**

Recording of WAV, OPUS, aacLc and other files fails silently

**To Reproduce**

1. Create a new Recorder
2. create a new RecordingConfig
3. Start recording with config
5. try to listen to recorder.onAmplitudeChanged
6. stop recording
7. try to listen to recorded file

**Expected behavior**

creating a useful recording file

**Additional context**

the amplitude is current:-infinity max: -160
It is working on 1.2.4 without any issues
Other devices we tested work without any issues
No error in console

**Comment by llfbandit** [comment]
Thanks for the report.
Using wav, opus or aac encoders fails the same way on those devices?
Are they on the same Android version?

**Comment by llfbandit** [comment]
I made an attempt to fix your issue in `compat_effect_issue_393` branch, are you able to test it?
To make it work, you should not use any effect like echo cancel, ...

**Comment by darajava** [comment]
@felixmede what gradle version are you using? I can't be sure yet since I can't reproduce the issue (The issue is happening for one of my users on a OnePlus, Android 11 too, which I can't emulate), but I think the issue might be related to upgrading to Gradle 8

I had a different issue caused by upgrading to Gradle 8 and adding the following to `gradle.properties` fixed it:

```
android.enableR8.fullMode=false
```

It might be the fix here too?

**Comment by darajava** [comment]
@felixmede @llfbandit  I can confirm that this issue is happening with one of my users on Android OnePlus 6T OxygenOS 11.1.1.2

Downgrading to:

```
  record: 5.1.0
  record_android: 1.2.4
```

fixes the issue

**Comment by felixmede** [comment]
It seems to be fixed for me with android_record 1.3.0
